src/_pytest/_code/code.py

```diff
@@ -262,7 +262,12 @@ def __str__(self) -> str:
             raise
         except BaseException:
             line = "???"
-        return "  File %r:%d in %s\n  %s\n" % (self.path, self.lineno + 1, name, line)
+        return "  File %r:%d in %s\n  %s\n" % (
```

Comment:
I think we should show the non-repr of the path here:

```suggestion
        return "  File %s:%d in %s\n  %s\n" % (
```

Reply:
I don't think so - Python also quotes the path in its tracebacks:

```
Traceback (most recent call last):
  File "test.py", line 1, in <module>
    raise Exception
```

Reply:
Ah, though Python uses `"` in any case:

```
  File "test"foo.py", line 1, in <module>
```

so I suppose we could do `File \"%s\"` here instead (and drop the `str(...)` again?).

Reply:
> I don't think so - Python also quotes the path in its tracebacks:

Ahh you are right. 👍 

> so I suppose we could do File \"%s\" here instead (and drop the str(...) again?).

Right on. 😁

Reply:
@The-Compiler

>  File "test.py", line 1, in <module>

The code seems to (before my oversight) do

> File "test.py":1 in <module>

Which looks kinda weird. Should we indeed change it to match the Python format, i.e `File "%s", line %d, in %s`?

Though that would break werkzeug as well, so not for this PR.

Reply:
Oh, I hadn't noticed that. I guess let's just stay with the previous format for now then?

Reply:
Actually I noticed previously this used `'` instead of `"`, based on the failure reported in https://github.com/pytest-dev/pytest/issues/7534:

```python
        traceback = "".join(str(line) for line in baz_exc.traceback)
>       assert "bb.py':1" in traceback  # a bit different than typical python tb
```